# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/jsapi/WebrtcGlobalInformation.h
# Commit: fb1a765ec213
# Full Hash: fb1a765ec213b12a3c6aa1474e8dc42585b1c2aa
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2022-12-21 21:21:23
# Regressor Bug: 1800136
# File Overlap Count: 1
# Description:
#   Bug 1800136: Fix gap between PC.close() and the storage of its final stats in PeerConnectionCtx. r=ng
#   
#   Ensures that:
#   1) We never have stats report in a still-registered PC _and_ the stats for
#      closed PCs.
# ==============================================================================

diff -r 3621f5bf1cb0 -r fb1a765ec213 dom/media/webrtc/jsapi/WebrtcGlobalInformation.h
--- a/dom/media/webrtc/jsapi/WebrtcGlobalInformation.h	Tue Dec 20 22:12:10 2022 +0000
+++ b/dom/media/webrtc/jsapi/WebrtcGlobalInformation.h	Tue Dec 20 22:12:10 2022 +0000
@@ -17,6 +17,7 @@
 class GlobalObject;
 class WebrtcGlobalStatisticsCallback;
 class WebrtcGlobalLoggingCallback;
+struct RTCStatsReportInternal;
 
 class WebrtcGlobalInformation {
  public:
@@ -42,7 +43,7 @@
   static bool AecDebug(const GlobalObject& aGlobal);
   static void GetAecDebugLogDir(const GlobalObject& aGlobal, nsAString& aDir);
 
-  static void StoreLongTermICEStatistics(PeerConnectionImpl& aPc);
+  static void StashStats(const RTCStatsReportInternal& aReport);
 
  private:
   WebrtcGlobalInformation() = delete;
